src/eth/GasEstimatorService.js

```diff
@@ -1,54 +1,114 @@
     this._multiplier = 1.55;
-    this._absolute = null;
+    this._transactionSpeed = 'fast';
+  }
+
+  initialize(settings) {
+    this._gasStationData = this.fetchGasStationData();
+
+    if (settings) {
+      this._parseConfig(settings.gasLimit, 'gasLimit');
+      this._parseConfig(settings.gasPrice, 'gasPrice');
+    }
+  }
+
+  _parseConfig(settings = 'default', label) {
+    return settings === 'default' || typeof settings === 'object'
+      ? this._setProperties(settings, label)
+      : (this[label] = settings);
   }
 
-  authenticate() {
-    const settings = this.get('web3').transactionSettings();
-    this._fallback = settings && settings.gasLimit
-      ? settings.gasLimit
-      : 4000000;
+  _setProperties(settings, label) {
+    if (settings === 'default') return;
+
+    return Object.keys(settings).map(key => {
+      const value = settings[key];
+      if (key === 'disable') {
+        this[
+          'disable' + label.charAt(0).toUpperCase() + label.slice(1)
+        ] = value;
+      } else {
+        this[key] = value;
+      }
+    });
+  }
+
+  async fetchGasStationData() {
+    try {
+      const response = await fetch(
+        'https://ethgasstation.info/json/ethgasAPI.json'
+      );
+      return response.json();
+    } catch (err) {
+      console.error('Error fetching gas data; disabling preset gas price');
+      this.disableGasPrice = true;
+    }
+  }
+
+  async getGasPrice(txSpeed) {
+    if (this.gasPrice) return this.gasPrice;
+    const speedSetting = txSpeed ? txSpeed : this.transactionSpeed;
+    const gasStationData = await this.gasStationData;
+
+    return gasStationData[speedSetting];
+  }
+
+  async getWaitTime(txSpeed) {
+    const speedSetting = txSpeed ? txSpeed : this.transactionSpeed;
+    const gasStationData = await this.gasStationData;
+
+    return gasStationData[`${speedSetting}Wait`];
   }
 
   async estimateGasLimit(transaction) {
+    if (this.gasLimit) return this.gasLimit;
+    if (this.disableGasLimit) return this.fallback;
+
     let web3Data = [];
     try {
       web3Data = await Promise.all([
         this.get('web3').getBlock('latest'),
         this.get('web3').estimateGas(transaction)
       ]);
     } catch (err) {
-      return this._fallback;
+      return this.fallback;
     }
 
     const blockLimit = web3Data[0].gasLimit;
     const estimate = web3Data[1];
 
     if (!this.multiplier && !this.absolute) {
-      return Math.min(this._absolute, blockLimit);
-    } else if (!this._absolute) {
-      return Math.min(parseInt(estimate * this._multiplier), blockLimit);
+      return Math.min(this.absolute, blockLimit);
+    } else if (!this.absolute) {
+      return Math.min(parseInt(estimate * this.multiplier), blockLimit);
     } else {
       return Math.min(
-        parseInt(estimate * this._multiplier),
-        this._absolute,
+        parseInt(estimate * this.multiplier),
+        this.absolute,
         blockLimit
       );
     }
   }
 
+  get gasStationData() {
```

Comment:
seems like this should just be an internal variable, no need for an external-facing getter because you would never want some other service to call it--also a bit clearer to name it ending in `Promise`